app/controllers/flights_controller.rb

```diff
@@ -8,20 +8,36 @@ def show
 
   end
 
-  def loggedin
-
-  end
-
   def search_flight
     to = flight_params[:to_id]
     from = flight_params[:from_id]
     dept_time  =  flight_params[:dept_time]
-    valid_airports(from, to)
-    search = Flight.new.get_flight(to, from , dept_time)
-    flash[:notice] = "No flights found. Please make another search." if search.empty?
+    if valid_airports(from, to).nil?
+      @flights_list = Flight.new.get_flight(to, from , dept_time)
+      if @flights_list.empty?
+        flash[:notice] = "No flights found. Please make another search."
+      else
+        return show_search(@flights_list)
+      end
+    end
     redirect_to root_url
   end
 
+  def show_search(list)
```

Comment:
Unused method argument - `list`. If it's necessary, use `_` or `_list` as an argument name to indicate that it won't be used. You can also write as `show_search(*)` if you want the method to accept any arguments but don't care about them.